### PR TITLE
[libgpg-error] fix arm64 build | [configure_make] fix arm host detection

### PR DESCRIPTION
**Pull request by autoantwort**
Previously failed with 
```
/bin/sh: ./mkerrcodes: No such file or directory
/bin/sh: ./mkheader: No such file or directory
make[2]: *** [gpg-error.h] Error 127
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```
[All logs](https://cloud.hilton.rwth-aachen.de/index.php/s/KdWjLc4miFYZQZJ)

The problem was that because of ` TARGET_ARCH: aarch64, BUILD_ARCH: arm64` the `--host` parameter was added and because of that `CC_FOR_BUILD` was set to `touch a.out | touch conftest | true` which does not produces real binaries.  
Now only `aarch64` is used to represent `arm64`

### CODE REVIEW

File: ports/libgpg-error/portfile.cmake [hunk 1]
```diff
@@ -29,7 +29,6 @@ if(VCPKG_CROSSCOMPILING)
 endif()
 
 vcpkg_configure_make(
-    AUTOCONFIG
```
Comment (dg0yt) [inline ports/libgpg-error/portfile.cmake:32]: This change is dangerous. Usually you have to add `DETERMINE_BUILD_TRIPLET` when you remove `AUTOCONFIG`, to fix cross-builds.

Comment (autoantwort) [reply ports/libgpg-error/portfile.cmake:32]: With `DETERMINE_BUILD_TRIPLET` the build fails, without the build is successful and even the cross build from arm64-osx to x64-osx works

Comment (dg0yt) [reply ports/libgpg-error/portfile.cmake:32]: The patch which deals with mkheader and mkerrorcode for cross builds actually relies on AUTOCONFIG.

**Comment by dg0yt** [comment]
> ~~~
> /bin/sh: ./mkerrcodes: No such file or directory
> /bin/sh: ./mkheader: No such file or directory

Cross builds are expected to use `$(HOST_TOOLS_PREFIX)/<tool>`, not `./<tool`. This is `cross-tools.patch` in connection with
https://github.com/microsoft/vcpkg/blob/d10d511f25620ca0f315cd83dcef6485efc63010/ports/libgpg-error/portfile.cmake#L27-L29

**Comment by dg0yt** [comment]
libgpg-error:arm64-osx builds fine on my x64 mac.

PR invalid.

**Comment by autoantwort** [comment]
Well it is ` TARGET_ARCH: aarch64, BUILD_ARCH: arm64`

**Comment by autoantwort** [comment]
@MonicaLiu0311 Can you review this?

**Comment by MonicaLiu0311** [comment]
`qca: arm64-windows` fails, but not related to this PR.

**Comment by autoantwort** [comment]
@MonicaLiu0311 Can you add https://github.com/microsoft/vcpkg/labels/info%3Areviewed ?

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**